pandas/_testing/asserters.py

```diff
@@ -578,13 +578,19 @@ def raise_assert_detail(
 
     if isinstance(left, np.ndarray):
         left = pprint_thing(left)
-    elif isinstance(left, (CategoricalDtype, NumpyEADtype, StringDtype)):
+    elif isinstance(left, (CategoricalDtype, NumpyEADtype)):
         left = repr(left)
+    elif isinstance(left, StringDtype):
+        # TODO(infer_string) this special case could be avoided if we have
+        # a more informative repr https://github.com/pandas-dev/pandas/issues/59342
+        left = f"StringDtype(storage={left.storage}, na_value={left.na_value})"
```

Comment:
See https://github.com/pandas-dev/pandas/pull/58451#discussion_r1696912714 for some discussion, but in short I am adding this for now because right now you can get test failures (like the ones that this PR is silencing with xfails) that have a very hard assert message like:

```
AssertionError: Attributes of DataFrame.iloc[:, 0] (column name="col1") are different

Attribute "dtype" are different
[left]:  string[pyarrow]
[right]: string[pyarrow]
```

which is not very helpful ...

The reason for that is because I did not bake the `pd.NA` vs `np.nan` information in the string alias / representation.

We need to resolve that issue more in general (see the linked issue in the comment above), but short term this ensures that at least for developing and running tests, we can distinguish easily between the different dtypes.